Adding Database

Albertamoahjunior opened the pull request:
I have connected the backend to a database now.  
Instead of using json file as the database, now postgre is been used as database.

Albertamoahjunior committed:
switched from using a json file as database to using postgre

Albertamoahjunior committed:
complete

genes2160 commented on backend/dbconfig.ts:
This needs to be removed into env

Albertamoahjunior replied:
Yes please.

genes2160 commented on backend/server.ts:
Why did you remove CORS ?

Albertamoahjunior replied:
I actually removed for testing purpose, to make sure the integration was working before I implement it again but I think I forgot to implement it again after I was done.

genes2160 commented on backend/src/controllers.ts:
Looks good

Albertamoahjunior replied:
Yes please

genes2160 reviewed (changes requested):
<img width="583" alt="image" src="https://github.com/user-attachments/assets/14f1f849-9440-4aeb-81ff-f54111d33967">

This indicates that adding a new task does not work i.e. The entries never make it into the db

Always ensure that the entries make way into the DB and debug errors you find
<img width="892" alt="image" src="https://github.com/user-attachments/assets/869a861b-812d-4218-81ac-09872f4fb25c">

genes2160 commented:
<img width="889" alt="image" src="https://github.com/user-attachments/assets/1d9c2044-b57d-4763-ba18-e639a511cf76">
Is this how the UI is intended ????

Albertamoahjunior committed:
changes to made to make the PR
- I beautified the add task dialog box a bit.
- I also made sure ids come from the database
- I am also env files now.

Albertamoahjunior committed:
fixed the database error correlation with frontend but with an issue, i doesn't work for the marking as done. Might have to change the entire logic

Albertamoahjunior committed:
made provision for creating database and table if they do not exist on the local machine

Albertamoahjunior committed:
wrapping overflow tasks and making their container scrollable

genes2160 commented on frontend/src/components/taskcontainer.tsx:
Next time delegate the 
```
http://localhost:2000
```
to its own file i.e. .env and import across so when the host changes there wont be a problem 

When it comes to this never hard code simply delegate to .env

Albertamoahjunior replied:
Yes please

genes2160 reviewed (approved):
Just few comments 
1. next time delegate a single .env that front and backend can make use of
i.e. structure can be 
|___.env (this ignored by git)
|____.env.example(this pushed to git)
|
|___frontend(folder)
|
|___backend(folder)


The both front and back can consume the .env
